### ISSUE TITLE
Completing Race 7 Before Unlocking Invitational Circuit Pass Doesn't Unlock Later

### ISSUE THREAD

**Issue by GFreak01** [wcolding/SWR_apworld#3]
completing race 7 (such that one of the invitational circuit races would normally unlock) does not unlock those races once the invitational circuit pass is received.

This causes the need to complete race 7 on the corresponding circuit a 2nd time to access.

**Comment by wcolding** [comment]
So the purpose of the invitational circuit pass is to replace the vanilla unlock methods for the invitationals. If you have the pass, you should be able to play Invitational 1, which should unlock Invitational 2 on completion and so on and so forth.

However, completing race 7 a second time should NOT be giving you a track unlock when using this unlock mode, so it does sound like there is indeed a bug.

**Comment by wcolding** [comment]
Also to clarify, the Invitational Circuit Pass (or 3rd progressive pass) won't necessarily be given on completing a circuit. It will be anywhere in the pool that logic allows

**Comment by GFreak01** [comment]
Sorry, perhaps I poorly explained the issue.   (perhaps this is actually a client issue instead of the world)

Steps Taken:
- I completed race 7 of the semi-pro circuit (which I agree shouldn't unlock the invitational race 3 until the pass has been received)
- The invitational pass is received
- Invitational race 3 was not accessible until I beat race 7 on semi-pro a 2nd time.

Settings:
progressive_circuits: false
invitational_circuit_pass: true

**Comment by wcolding** [comment]
No so the way the Invitational Pass is supposed to function is as follows:

1. You find Invitational Circuit Pass
2. Invitational 1 unlocks
3. You play Invitational 1
4. Invitational 2 unlocks
5. You play Invitational 2
6. Invitational 3 unlocks
7. You play Invitational 3
8. Invitataional 4 unlocks

It is SUPPOSED to completely replace the vanilla unlock system entirely if you are using it. What I am learning from your issue is that maybe there is a secondary function the game calls to write Invitational unlocks that I have missed, and should disable when using the Invitational Circuit pass.

In short, Invitational Circuit Pass is supposed to make it function like the first three circuits, but gates the whole circuit behind getting that item. If you play with Circuit Pass ***without*** Invitational then you will have passes for Semi-Pro and Galactic and the Invitational unlocks WILL be their vanilla methods.
